addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -262,10 +262,15 @@ def _get_random_operator(self, lang=None, country_id=None):
         active_channels = self.env.cr.dictfetchall()
         operator = None
         # Try to match an operator with the same lang as the visitor
+        # If no operator with the same lang, try to match an operator with the preferred lang
         if lang:
             same_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: operator.partner_id.lang == lang)
             if same_lang_operator_ids:
                 operator = self._get_less_active_operator(active_channels, same_lang_operator_ids)
+            else:
+                preferred_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: lang in operator.livechat_langs.mapped('code'))
```

Comment:
[`mapped` is performed automatically when you access the property of a relational field](https://github.com/odoo/odoo/blob/11f75dd64d5841eb28912c3098a392bfe93ed63c/odoo/fields.py#L2800).
```suggestion
                preferred_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: lang in operator.livechat_langs.code)
```